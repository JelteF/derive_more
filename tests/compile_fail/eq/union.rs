#[derive(derive_more::Eq)]
union IntOrFloat {
    i: u32,
}

impl PartialEq for IntOrFloat {
    fn eq(&self, _: &Self) -> bool {
        unimplemented!()
    }
}

fn main() {}
