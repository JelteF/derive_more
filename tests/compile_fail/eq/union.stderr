error: `Eq` cannot be derived for unions
 --> tests/compile_fail/eq/union.rs:2:1
  |
2 | union IntOrFloat {
  | ^^^^^
